solutions/CF Orac-LCM.mdx

```diff
@@ -25,22 +25,61 @@ ll a,b;
 int n;
 
 void red() { // restore invariant mentioned above
-	ll g = __gcd(a,b);
-	tie(a,b) = make_pair(g,a/g*b);
+    ll g = __gcd(a,b);
+    tie(a,b) = make_pair(g,a/g*b);
 }
 
 int main() {
-	cin >> n >> a >> b;
-	red();
-	for (int i = 2; i < n; i++) {
-		ll x; cin >> x; 
-		b = __gcd(b,x);
-		red();
-	}
-	cout << b << endl;
+    cin >> n >> a >> b;
+    red();
+    for (int i = 2; i < n; i++) {
+        ll x; cin >> x; 
+        b = __gcd(b,x);
+        red();
+    }
+    cout << b << endl;
 }
 ```
 
 If it's hard to understand what exactly this code is doing at first glance, a good first step is to simulating the code in the case where each $a_i$ is a power of the same prime (say, $2^k$). If the algorithm works for this case, then it will also work in the general case since the contributions of different primes are computed independently and multiplied together.
 
 <!-- the GCD function is [multiplicative](https://en.wikipedia.org/wiki/Multiplicative_function). -->
+
+## Solution: GCD/LCM
+Consider fixing $j$. How can the gcd of all lcms of nums[$j$] and all nums[$i$] (such that $i$ is in range $[0,j]$) be quickly computed? The result would be nums[$j$]$*C$, where $C$ is the gcd of all nums[$i$], and we can keep track of this gcd on the run.
```

Comment:
I'd prefer if we used array `a` instead of `nums` as described in the problem statement.

Some wording changes might help too:

* "Consider fixing an index $j$" ...
* ... and all numbers $\text{a}[i]$ ($1 \leq i < j$) ...
* ... where $C$ is $\text{GCD}(\left{ a_1, a_2, \dots, a_{j-1} \right})$ ...

Reply:
Thank you. I will make those edits.